cookbook/core/type_system/annotation.py

```diff
@@ -0,0 +1,42 @@
+"""
+.. _annotations:
+
+Parameter Annotations
+--------------------------
+
+This example explains how to annotate parameters within Flyte. These
+annotations are supposed to be semantic descriptions of types and are
+accessible from the serialized flyteidl LiteralType message. They arenot
+accesible at runtime within a task.
+
+"""
+import typing
+
+from flytekit import task, workflow
+
+# %%
+# Flytekit has an object to hold annotations.
+from flytekit.core.annotation import FlyteAnnotation
+
+# %%
+# FlyteAnnotation can hold arbitrary data objects and should be stuffed inside
+# the `typing.Annotated` type guard.
+typing.Annotated[int, FlyteAnnotation({"some": "data"})]
```

Comment:
I've not looked at the flytekit PR but why is it that we require `FlyteAnnotation` as in what happens if I do:
```python
typing.Annotated[int, ValueRange(3, 10), ctype("char")]
```

Will we error? wrap it in FlyteAnnotation? drop it on the floor?

cc @eapolinario